### ISSUE TITLE
[Quickfort] Track carving - rotation issue

### ISSUE THREAD

**Issue by fricy** [DFHack/dfhack#3344]
Rotating or flipping track carving layouts doesn't preserve the intended track direction.

Blueprint:
```
"#dig start(5,4) Stage 3/2a: Carve downstream exit",,,,,,,,,
,,,,,,,,,#
,,,,,trackSE2,trackSW2,,,#
,,,trackEW2,trackEW2,trackNW2,trackNS2,trackEW2,trackEW2,#
,,trackNS2,,,,trackNS2,,,#
trackEW2,trackEW2,trackNE2,trackEW2,trackEW2,trackEW2,,,,#
,,,,,,,,,#
,,,,,,,,,#
```

Intended track layout:
![carve_intended](https://user-images.githubusercontent.com/5114750/235752035-42b0b330-02e7-4652-bfdc-870f8ba7c609.png)

Result of 180' rotation:
![carve_rotated](https://user-images.githubusercontent.com/5114750/235752246-21a13288-4163-4c52-bd8b-aabb2fb34591.png)

**Comment by myk002** [comment]
good catch. I ensured built tracks were properly rotated, but I seem to have missed carved. Will fix.

**Comment by fricy** [comment]
One more quickfort issue if you're at it:
I can't build horizontal axles over stairs. I can manually build them as long as they have a gear supporting them next to the stair, but not in quickfort.

**Comment by myk002** [comment]
Thanks! Could you make a separate issue for that? it's in a very different part of the code from this one.

**Comment by fricy** [comment]
Sorry, I never know which issue deserves a new number...

**Comment by myk002** [comment]
So the logic for transforming track designations is completely implemented for constructed tracks, but dig blueprints need to get hooked up to the transformation logic like build blueprints are. this will take a bit of refactoring.

The amusing thing is that I successfully tested transforming carved tracks, but the difference was that the tracks I tested were expressed using expansion syntax (e.g. `T(10x5)`), which is transformed at an earlier stage (during parsing). directional carved tracks that are specified tile by tile never got tested, so it never got implemented.

**Comment by fricy** [comment]
T() designation indeed seem to work. And not that hard to figure out....

Except they don't seem to accept priority T(4x1)2 and carving at 4 takes forever... I bet you like my constatant nitpicking. :))

On that note: Are dumping commands broken until the rewrite? I tried adding db(7x5) to #dig and #build both, but it's rejected as invalid command.

I read a few pages of you bay12 development thread for this util, great job by the way!

**Comment by myk002** [comment]
> Except they don't seem to accept priority `T(4x1)2`

The syntax for that would be `T2(4x1)` -- the expansion syntax portion always comes at the end

> I tried adding `db(7x5)` to #dig and #build both, but it's rejected as invalid command.

Dumping would be `bd(7x5)` and would go in a `#dig` blueprint. Now that the hotkeys aren't "obvious" anymore, I really need to document what keys mean what.

> I read a few pages of you bay12 development thread for this util, great job by the way!

Thanks!

**Comment by myk002** [comment]
that's fair, we could hide "invalid" dump tiles just like we do for chopping where there are no trees. Could you file a separate issue for that?

**Comment by fricy** [comment]
[I'd appreciate your opinion when you have the time.](https://docs.google.com/spreadsheets/d/1VuPxJFhuCMA8l57dqU9XwxFA0xmjnyni/edit?usp=sharing&ouid=103483780848029114039&rtpof=true&sd=true)

**Comment by myk002** [comment]
That blueprint looks very useful! There were a couple points that I thought could use clearer explanations in the docs:
1. can exits be done at any level or just at the bottom/top?
2. where are the potential tiles for pressure plates, specifically?
3. I know this needs to be hooked up to external power, but that is never stated explicitly

also, the center tile offsets in stage 3 appear to be off by 1

Would you like to include this blueprint in the quickfort blueprint library? You'd retain ownership of the online files for continued updates. The version included with DFHack would be a cvs export of the xlsx, but we'd link to the online version from the library guide -- just like Dreamfort.

**Comment by fricy** [comment]
Thx for the feedback, that's exactly what I was looking for! Fresh eyes are usefull.

Yeah, I guess eventually it should be part of the library. But it's not there yet, and building it is a bit clunky until some of the bugs I filed are fixed. And full potential would need the pressure plate extensions I posted yesterday.

First I'll post it to reddit for some more feedback, and testing. I'm not even sure this is the best design, I have another that's slightly slower, but more flexible full symetric 5x5 version. This is the compact with compromises.
